cirq-core/cirq/ops/common_gates_test.py

```diff
@@ -1116,3 +1116,170 @@ def test_approx_eq():
     assert not cirq.approx_eq(cirq.Y**0.1, cirq.Y**0.2, atol=0.05)
     assert cirq.approx_eq(cirq.X**0.1, cirq.X**0.2, atol=0.3)
     assert not cirq.approx_eq(cirq.X**0.1, cirq.X**0.2, atol=0.05)
+
```

Comment:
Could you add tests demonstrating how a dim-3 qudit interacts with dim-2 X/Z, and vice versa?

Reply:
It just fails `.on(wrong_shape_qubit)` because GateOperation checks the shapes match. However I added a test.

Reply:
Also note @cduck has a `SingleQubitSubspaceGate` that would allow dim-K gates to be applied to a subspace of a dim-N qubit where N >= K (https://github.com/cduck/cirqtools/blob/2fbff43fa945dfa6155e83f4a72c16774ca26e42/cirqtools/qudit/common_gates.py#L85). That applies to any gate, not just these X/Z additions. However it sounds like that's not on the roadmap (https://github.com/quantumlib/Cirq/issues/3190#issuecomment-1111396753).